sentry_sdk/spotlight.py

```diff
@@ -52,50 +61,137 @@ def capture_envelope(self, envelope):
             req.close()
         except Exception as e:
             self.tries += 1
-            logger.warning(str(e))
+            sentry_logger.warning(str(e))
 
 
 try:
-    from django.http import HttpResponseServerError
+    from django.utils.deprecation import MiddlewareMixin
+    from django.http import HttpResponseServerError, HttpResponse, HttpRequest
     from django.conf import settings
 
-    class SpotlightMiddleware:
-        def __init__(self, get_response):
-            # type: (Any, Callable[..., Any]) -> None
-            self.get_response = get_response
-
-        def __call__(self, request):
-            # type: (Any, Any) -> Any
-            return self.get_response(request)
+    SPOTLIGHT_JS_ENTRY_PATH = "/assets/main.js"
+    SPOTLIGHT_JS_SNIPPET_PATTERN = (
+        '<script type="module" crossorigin src="{}"></script>'
+    )
+    SPOTLIGHT_ERROR_PAGE_SNIPPET = (
+        '<html><base href="{spotlight_url}">\n'
+        '<script>window.__spotlight = {{ initOptions: {{ fullPage: true, startFrom: "/errors/{event_id}" }}}};</script>\n'
+    )
+    CHARSET_PREFIX = "charset="
+    BODY_CLOSE_TAG = "</body>"
+    BODY_CLOSE_TAG_POSSIBILITIES = [
+        "".join(chars)
+        for chars in product(*zip(BODY_CLOSE_TAG.upper(), BODY_CLOSE_TAG.lower()))
+    ]
```

Comment:
```suggestion
    BODY_CLOSE_TAG_POSSIBILITIES = {
        "".join(chars)
        for chars in product(*zip(BODY_CLOSE_TAG.upper(), BODY_CLOSE_TAG.lower()))
    }
```
There's a lot of duplicates in that list otherwise